custom-metrics-stackdriver-adapter/pkg/adapter/translator/query_builder.go

```diff
@@ -276,10 +397,45 @@ func (qb QueryBuilder) getFilterBuilder() utils.FilterBuilder {
 		return utils.NewFilterBuilder(utils.SchemaTypes[utils.PrometheusSchemaKey])
 	}
 
+	// node type FilterBuilder
+	if qb.namespace == "" {
+		return utils.NewFilterBuilder(utils.SchemaTypes[utils.NodeSchemaKey])
+	}
+
 	// pod type FilterBuilder
 	return utils.NewFilterBuilder(utils.SchemaTypes[utils.PodSchemaKey])
 }
 
+// composeFilter is an internal helper function to compose filter criteria
+// when namespace is NOT empty
+func (qb QueryBuilder) composeFilter() string {
+	filterBuilder := qb.getFilterBuilder().
+		WithMetricType(qb.metricName).
+		WithProject(qb.translator.config.Project).
+		WithCluster(qb.translator.config.Cluster)
+
+	resourceNames := qb.getResourceNames()
+	if qb.translator.useNewResourceModel {
+		// new resource model specific filters
+		filterBuilder = filterBuilder.WithLocation(qb.translator.config.Location)
+		if !qb.nodes.isNodesContainerEmpty() {
+			// node metrics
+			filterBuilder = filterBuilder.WithNodes(resourceNames)
```

Comment:
Can we just return 
```
filterBuilder = filterBuilder.WithNodes(resourceNames)
return filterBuilder.Build()
```
to reduce the usage of if else?

Reply:
sure